core/src/main/java/org/ehcache/core/internal/service/ServiceLocator.java

```diff
@@ -425,9 +438,28 @@ ServiceMap lookupDependenciesOf(ServiceMap resolved, Class<? extends Service> re
         }
       }
     }
-    OptionalServiceDependencies optionaAnnotation = clazz.getAnnotation(OptionalServiceDependencies.class);
-    if (optionaAnnotation != null) {
-      for (String className : optionaAnnotation.value()) {
+
+    for (Class<?> interfaceClazz : clazz.getInterfaces()) {
+      if (Service.class.isAssignableFrom(interfaceClazz)) {
+        dependencies.addAll(identifyImmediateDependenciesOf(Service.class.getClass().cast(interfaceClazz)));
+      }
+    }
+
+    dependencies.addAll(identifyImmediateDependenciesOf(clazz.getSuperclass()));
+
+    return dependencies;
+  }
+
+  private static Set<Class<? extends Service>> identifyImmediateOptionalDependenciesOf(final Class<?> clazz) {
+    if (clazz == null) {
```

Comment:
Why would clazz be null?

Reply:
Go figure. It's present in `identifyImmediateDependenciesOf` so I kept it in `identifyImmediateOptionalDependenciesOf` too. Better be safe than sorry.

Reply:
Hum... Recursion I guess